classes/local/api.php

```diff
@@ -185,7 +185,9 @@ private function add_postname($storedfile, $key) {
             throw new \moodle_exception('needphp55orhigher', 'tool_opencast');
         }
 
-        $filename = $storedfile->get_source();
+        $sourse = @unserialize($storedfile->get_source());
+        $filename = $sourse ? $sourse : $storedfile->get_source();
```

Comment:
I feel that this needs an explanatory comment that describes what is done here. I do understand it now (either unserialise the object and retrieve a filename from its field, or use the value prior to serialisation as a filename which only happens if it is not an object).

Moreover, instead of relying on `$source ?` evaluating correctly, please rather use `is_object($source)` instead, which also is way cleaner to read for a human.

Note that there is a typo in `$sourse`.